sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala

```diff
@@ -77,11 +77,16 @@ case class Average(child: Expression) extends DeclarativeAggregate with Implicit
   )
 
   // If all input are nulls, count will be 0 and we will get null after the division.
-  override lazy val evaluateExpression = child.dataType match {
-    case _: DecimalType =>
-      DecimalPrecision.decimalAndDecimal(sum / count.cast(DecimalType.LongDecimal)).cast(resultType)
-    case _ =>
-      sum.cast(resultType) / count.cast(resultType)
+  // We can't directly use `Divide` as it throws an exception under ansi mode.
```

Comment:
can we put `failOnError` as a constructor parameter in `Divide`, like what @leanken did to the statistical aggregate functions? Then here we just need to write `Divide(sum.cast(resultType), count.cast(resultType), failOnError = false)`

Reply:
done